src/main/java/org/candlepin/controller/ContentAccessManager.java

```diff
@@ -906,8 +957,8 @@ public AnonymousContentAccessCertificate getCertificate(AnonymousCloudConsumer c
             }
 
             payload = createAnonPayloadAndSignature(products);
-            List<ContentInfo> contentInfo = getContentInfo(products);
-            content = convertContentInfoToContentDto(contentInfo);
+            Map<ContentInfo, Boolean> contentInfo = getContentInfo(products);
```

Comment:
Were we worried at all about the collision issue here as well?

Reply:
I think we should be, if we're worried about the other case. But here it will have to be translated all the way into the model.dto.Content. I'll update it

Reply:
Looking at this again, we don't really need the enabled/disabled flag here. In fact, from the `dto.Content` objects we're building in order to get into the x509 part of the cert, we only ever use the `path` field from them.
However, we should still translate them into a Map of Content ID to ProductContent, to be consistent with the list of content we have in the payload. So in short, I'm changing this to:
- Translate ProductInfo into Map<String, ProductContent>
- Use that map to build the payload / content list part of the cert.
- Translate the ProductContent from that same map into dto.Content objects that only have the `path` field populated, and throw that into the X509 part of the cert.